packages/eslint-plugin/src/rules/no-unused-vars.ts

```diff
@@ -419,10 +419,23 @@ export default createRule<Options, MessageIds>({
                 ref.from.variableScope === unusedVar.scope.variableScope,
             );
 
+            const id = writeReferences.length
```

Comment:
Even though these branches collapse to do the same thing, I think it could be good to test both branches, in case one day they get separated (for example, to report different error messages)

What do you think about adding test cases with all 4 report loc fields for some cases like the following?
```ts
let foo: number;
```
```ts
let foo: number = 1;
foo = 2;
foo = 3;
```